xonsh/dirstack.py

```diff
@@ -4,12 +4,100 @@
 import glob
 import argparse
 import builtins
+import subprocess
 
 from xonsh.lazyasd import lazyobject
 from xonsh.tools import get_sep
+from xonsh.platform import ON_WINDOWS
 
 DIRSTACK = []
 """A list containing the currently remembered directories."""
+_unc_tempDrives = {}
+""" drive: sharePath for temp drive letters we create for UNC mapping"""
+
+
+def _unc_check_enabled()->bool:
+    """Check whether CMD.EXE is enforcing no-UNC-as-working-directory check.
+    Oh noes! can the registry entry be defined in HKCU too?
```

Comment:
Well, in my case it was in HKLM.... We will have to see if this cause trouble for people